lib/model/message_list.dart

```diff
@@ -86,6 +88,68 @@ class MessageListView extends ChangeNotifier {
     notifyListeners();
   }
 
+  applyChangesToMessage(UpdateMessageEvent event, Message message) {
+    if (event.renderingOnly != null && event.renderingOnly == true) {
```

Comment:
Huh, that's interesting - what's your policy on backwards compatibility? Do all clients you release work with all server versions?

Reply:
Yeah, the self-hosting makes it necessary for the client to have some window of backward compatibility.

For this client, the current window is described in the readme file:
```
We support Zulip Server 4.0 and later.  For API features added in
newer versions, use `TODO(server-N)` comments (like those you see
in the existing code.)
```

(More generally there's discussion on [a "Release lifecycle" page](https://zulip.readthedocs.io/en/latest/overview/release-lifecycle.html) in our developer docs.)

We'll probably bump that version higher sometime soon. But until we do that explicitly, I'd like to keep the code uniformly supporting the version we say we're supporting. That helps keep it manageable to be able to look at a piece of code and then at the API docs and confirm the code is behaving as it should, and to avoid accidentally leaving out something we need on newer versions.

Reply:
OK, I implemented this behavior and threw in an extra unit test to check it.